src/diffusers/models/modeling_utils.py

```diff
@@ -859,6 +977,76 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
 
         return model
 
+    # Adapted from `transformers`.
+    @wraps(torch.nn.Module.cuda)
+    def cuda(self, *args, **kwargs):
+        # Checks if the model has been loaded in 4-bit or 8-bit with BNB
+        if getattr(self, "quantization_method", None) == QuantizationMethod.BITS_AND_BYTES:
+            if getattr(self, "is_loaded_in_8bit", False):
+                raise ValueError(
+                    "Calling `cuda()` is not supported for `8-bit` quantized models. "
+                    " Please use the model as it is, since the model has already been set to the correct devices."
+                )
+            elif is_bitsandbytes_version("<", "0.43.2"):
+                raise ValueError(
+                    "Calling `cuda()` is not supported for `4-bit` quantized models with the installed version of bitsandbytes. "
+                    f"The current device is `{self.device}`. If you intended to move the model, please install bitsandbytes >= 0.43.2."
+                )
```

Comment:
I have taken the liberty to pin `bitsandbytes` version to 0.43.3 as we're introducing it for the first time. So, if that is the case, I think we can safely remove these? 

https://github.com/huggingface/diffusers/blob/859f2d76bdccf28e3ae35660c60cd9df946b197c/src/diffusers/quantizers/bitsandbytes/bnb_quantizer.py#L76

Reply:
Yes !